src/cli/cbenv_register.rs

```diff
@@ -207,42 +205,28 @@ fn clusters_register(
     ))
 }
 
-pub fn update_config_file(guard: &mut MutexGuard<State>, span: Span) -> Result<(), ShellError> {
-    let path = match guard.config_path() {
-        Some(p) => p,
-        None => {
-            return Err(generic_error(
-                "A config path must be discoverable to save config",
-                None,
-                span,
-            ));
-        }
-    };
-    let mut cluster_configs = Vec::new();
-    for (identifier, cluster) in guard.clusters() {
-        cluster_configs.push(ClusterConfig::from((identifier.clone(), cluster)))
-    }
-    let mut capella_configs = Vec::new();
-    for (identifier, c) in guard.capella_orgs() {
-        capella_configs.push(CapellaOrganizationConfig::new(
-            identifier.clone(),
-            c.secret_key(),
-            c.access_key(),
-            Some(c.timeout()),
-            c.default_project(),
+fn save_new_cluster_config(
+    guard: &mut MutexGuard<State>,
+    span: Span,
+    identifier: String,
+) -> Result<(), ShellError> {
+    let mut config = read_config_file(guard, span)?;
+    let clusters = config.clusters_mut();
+
+    if clusters.iter().any(|c| c.identifier() == identifier) {
```

Comment:
This will prevent users from updating an existing cluster,is that definitely what we want?

Reply:
Yes since the register/unregister commands aren't intended to be used to update cluster information, only register new clusters. If you called register on an existing cluster it would error before reaching this point.